lib/components/component.dart

```diff
@@ -89,4 +98,43 @@ abstract class Component {
 
   /// Get a list of non removed effects
   List<ComponentEffect> get effects => _effectsHandler.effects;
+
+  /// Uses the game passed in to prepare the child component before it is added
+  /// to the list of children
+  void addChild(Game gameRef, Component c) {
+    if (gameRef is BaseGame) {
```

Comment:
Since this method has this logic, it bothers me that the `children` field is public.

Since it is not private, people will be able to just call `children.add`, skipping this logic and probably creating some confusion.

My suggestion, keep `children` private and create some sort of method here that lets the children be iterated, like a `void forEachChild(Function(Component))`.

What do you think?

Reply:
I agree that `children` shouldn't be public, it is just a bit ugly to reimplement proxy methods for so many of the collection methods on the component imho. But if we don't come up with something smarter that will be the way we will have to go.

Reply:
you don't have to, make a method that returns a copy. you just need to provide that + add and maybe addAll if you want

Reply:
And `removeChild` would have to be added too, but yes, maybe a copy would be nicer than a `forEachChild` method.